mechanicalsoup/form.py

```diff
@@ -303,33 +305,41 @@ def choose_submit(self, submit):
             form.choose_submit('form_name_attr')
             browser.submit_selected()
         """
+        # Since choose_submit is destructive, it doesn't make sense to call
```

Comment:
Actually, I'd say this is a sign that `choose_submit` does not do the right thing. It's not documented to be destructive, and shouldn't be actually.

This `name` attribute stripping is in the wrong place IMHO. We should just get rid of this `del inp['name']` and ignore the corresponding elements in `_request`. Or did I miss anything?